components/ntp_background_images/browser/ntp_background_images_data.h

```diff
@@ -69,7 +69,7 @@ struct Background {
 struct NTPBackgroundImagesData {
   NTPBackgroundImagesData();
   NTPBackgroundImagesData(const std::string& json_string,
-                          const base::FilePath& installed_dir);
+                          const base::FilePath& installed_dir, bool is_sponsored_image);
```

Comment:
How about creating new struct for NTP BI?
Using same struct for SI and BI makes this complex.

Reply:
I think using same struct for NTP SI and NTP SR seems fine because SR uses whole SI spec.